### PR TITLE
Automated test.

### PR DESCRIPTION
**Pull request by patrycja93**
Created four winner's condition sequences for automated tests. Change the way of running the main class of the project(added arguments).

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -6,13 +6,41 @@ Author: Patrycja Hyjek
...
+
+For example `java -jar ./target/ox-1.0.0.jar 5 4 0`
 Results from demo will be saved in: main/resources/pl/patrycja/ox/ui/test_out.txt
 
-4. Launch game:
-* `mvn clean install`
-* `java -jar ./target/ox-1.0.0.jar`
+### Tips:
+
+1. If you put the second argument greater than first in demo mode, argument will be switched.
```
Comment (BulaMar) [inline readme.md]: arguments will be switched

File: src/main/java/pl/patrycja/ox/GameSettings.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public GameSettingsBuilder ui(UI ui) {
         /**
          * This method returns object GameSettings
          */
-        public GameSettings build() {
+        public GameSettings build() throws IllegalStateException{
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/GameSettings.java]: Do we need it here? Is it Runtime?

File: src/main/java/pl/patrycja/ox/gameflow/Game.java [hunk 3]
```diff
@@ -2,28 +2,27 @@
...
+    private GameSettings gameSettings;
+
     private Player firstPlayer = new Player("A", Sign.CROSS);
     private Player secondPlayer = new Player("B", Sign.NAUGHT);
-    private GameSettings gameSettings;
 
-    public Game(UI ui) {
-        this.ui = ui;
-        this.gameSettings = getGameSettings(ui);
+    public Game(GameSettings gameSettings) {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/Game.java]: 👍

File: src/main/java/pl/patrycja/ox/gameflow/Game.java [hunk 4]
```diff
@@ -2,28 +2,27 @@
...
+        this.gameSettings = gameSettings;
     }
 
-    public void play() {
+    void play() {
         gameSettings.setPlayer();
-        PlayerChanger playerChanger = new PlayerChanger(List.of(firstPlayer, secondPlayer), gameSettings);
+        PlayerChanger playerChanger = new PlayerChanger(List.of(this.firstPlayer, secondPlayer), gameSettings);
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/Game.java]: Use this keyword for both or for none.

File: src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java [hunk 5]
```diff
@@ -12,18 +15,85 @@
...
 
         boolean ifDemo = checkIfDemo(args);
-        UI ui = UIFactory.set(ifDemo);
 
-        Game game = new Game(ui);
+        Game game = setMode(args, ifDemo);
         game.play();
     }
 
+    private static Game setMode(String[] args, boolean ifDemo) {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java]: We are rather creating here sth then setting don't U think?

File: src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java [hunk 6]
```diff
@@ -12,18 +15,85 @@
...
+            gameSettings = GameSettings.builder()
+                    .ui(UIFactory.set(false))
+                    .build();
+        }
+        return new Game(gameSettings);
+    }
+
     private static boolean checkIfDemo(String[] args) {
-        return args.length > 0 && args[0].equals("demo");
+        return args.length == 3;
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java:25]: Why 3? Maybe `REQUIRED_NUMBER_OF_ARGUMENTS`?

File: src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java [hunk 7]
```diff
@@ -12,18 +15,85 @@
...
+    }
+
+    private static GameSettings getGameSettings(String[] args) {
+        GameSettings.GameSettingsBuilder gameSettings = GameSettings.builder();
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java]: You do not need that, this is static so you have access wherever you want to use it.

File: src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java [hunk 8]
```diff
@@ -12,18 +15,85 @@
...
+        if (validateArguments(boardSize, unbrokenLine)) {
+            setBoardSizeAndUnbrokenLine(gameSettings, boardSize, unbrokenLine, towards);
+        }
+
+        return gameSettings
+                .ui(UIFactory.set(true))
+                .build();
+    }
+
+    private static void setBoardSizeAndUnbrokenLine(GameSettings.GameSettingsBuilder gameSettings, String boardSize, String unbrokenLine, String towards) {
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java]: Do not pass builder, just use it.

File: src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java [hunk 9]
```diff
@@ -12,18 +15,85 @@
...
+                for (int i = 1; i <= (size - line); i++) {
+                    counter += 2 * (((size - i) - line) + 1);
+                }
+                return counter;
+            }
+            default:
+                return 3;
+        }
     }
 }
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java:27]: This demo/automated test things should be extracted to separate method.

File: src/main/java/pl/patrycja/ox/ui/Sequence.java [hunk 10]
```diff
@@ -0,0 +1,143 @@
+package pl.patrycja.ox.ui;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Sequence {
+
+    private String fileSequence = "src/main/resources/pl/patrycja/ox/ui/test_sequence.txt";
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/ui/Sequence.java:10]: `getClass().getResource()` please

File: src/main/java/pl/patrycja/ox/ui/Sequence.java [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+        try {
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file));
+            bufferedWriter.write(input.toString());
+            bufferedWriter.append("\n");
+            bufferedWriter.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
```
Comment (BulaMar) [inline src/main/java/pl/patrycja/ox/ui/Sequence.java:143]: Got it that logic is complicated for generating sequences but can we somehow mark those additional `1`s.
Also please extract repetable things to variables and reuse them.

File: src/main/resources/pl/patrycja/ox/ui/test_out.txt [hunk 12]
```diff
@@ -1,101 +1,229 @@
...
-X O X 
-O 8 9 
+ 1  2  3  4  5 
+ 6  O  X  9 10 
+11 12  O  X 15 
+16 17 18  O  X 
+21 22 23 24  O 
 
-Winner is O.
-End game!
```
Comment (BulaMar) [inline src/main/resources/pl/patrycja/ox/ui/test_out.txt:101]: Maybe add this to .gitignore?

File: src/main/resources/pl/patrycja/ox/ui/test_sequence.txt [hunk 13]
```diff
@@ -0,0 +1 @@
+X 4 5 8 9 12 13 16 5 4 9 10 13 14 17 9 10 13 14 17 18 21 10 9 14 15 18 19 22
```
Comment (BulaMar) [inline src/main/resources/pl/patrycja/ox/ui/test_sequence.txt]: This as well.

**Review by BulaMar — CHANGES_REQUESTED** [review]
Not much but still need some work

**Comment by patrycja93** [comment]
I made corrections.

**Review by BulaMar — APPROVED** [review]
Thank U for changes.

**Pull request merged**